server/user.go

```diff
@@ -21,27 +23,109 @@ func (p *Plugin) getUserFromEmail(email string) (string, error) {
 	return matches[1], nil
 }
 
+func (p *Plugin) makeJoinUrl(meetingUrl string) string {
+	return strings.Replace(meetingUrl, "webex.com/meet/", "webex.com/join/", 1)
+}
+
+func (p *Plugin) makeStartUrl(meetingUrl string) string {
+	return strings.Replace(meetingUrl, "webex.com/meet/", "webex.com/start/", 1)
+}
+
+func (p *Plugin) getEmailAndEmailName(mattermostUserId string) (string, string, error) {
+	user, appErr := p.API.GetUser(mattermostUserId)
+	if appErr != nil {
+		p.errorf("error getting mattermost user from mattermostUserId: %s", mattermostUserId)
+		return "", "", errors.New("error getting mattermost user from mattermostUserId, please contact your system administrator")
+	}
+	emailName, err := p.getUserFromEmail(user.Email)
+	if err != nil {
+		return "", "", err
+	}
+
+	return user.Email, emailName, nil
+}
+
+func (p *Plugin) getUrlFromRoom(mattermostUserId string) (string, error) {
+	userInfo, err := p.store.LoadUserInfo(mattermostUserId)
+	if err == ErrUserNotFound {
+		return "", err
+	}
+	if err != nil {
+		// unexpected error
+		p.errorf("error from the store when retrieving room for mattermostUserId: %s, error: %v", mattermostUserId, err)
+		return "", errors.New(fmt.Sprintf("error getting your room from the store, please contact your system administrator. Error: %v", err))
```

Comment:
I'm assuming the KV Store?  The user and possibly even sysadmin wouldn't know what that is.  What would they need to do?

Reply:
@aaronrothschild It would be rare that this would happen. They would probably end up contacting us, and if they said "there was a problem with the store" then we would know what they were talking about. Otherwise, we wouldn't know to look there. I'm open to suggestions of what else to put there. It's not an error that they sys admin could fix.